teps/0115-tekton-catalog-reorganization.md

```diff
@@ -0,0 +1,288 @@
+directory structure with two `Tasks` (argocd and golang-build) and one `Pipeline` (go-release) is organized as such:
+
+```shell
+./task/
+  /argocd
+    /0.1
+      /README.md
+      /argocd.yaml
+      /samples
+      /tests
+    /0.2
+      /README.md
+      /argocd.yaml
+      /samples
+      /tests
+    /OWNERS
+  /golang-build
+    /0.1
+      /README.md
+      /golang-build.yaml
+      /samples
+      /tests
+    /0.2
+      /README.md
+      /golang-build.yaml
+      /samples
+      /tests
+    /OWNERS
+./pipeline/
+  /go-release
+    /0.1
+      /README.md
+      /go-release.yaml
+      /samples
+      /tests
+    /0.2
+      /README.md
+      /go-release.yaml
+      /samples
+      /tests
+    /OWNERS
+```
+
+For further information, see [TEP-0003: Tekton Catalog Organization][tep-0003].
+
+The current directory-based organization presents challenges:
+- Bumping versions of resources is cumbersome which encourages minor making changes in existing versions. Some of these 
+changes that seem minor lead to surprising breaking behavior for users, as described in [tektoncd/catalog#784][784].
+- Resources that are frequently updated with new versions end up with a lot of duplications because any change requires
+copying over everything into a new directory. For example, [git-clone][git-clone] `Task` has seven versions currently
+with a lot of duplication across versions.
+
+Previously, there was tight coupling between the organization and how users referenced resources which made it hard to 
+change the organization of the Catalog without breaking users. In [TEP-0110][tep-0110], we proposed decoupling Catalog
+organization from resource reference. Now that [TEP-0110][tep-0110] is implemented, we can reorganize the Catalog to 
+address the challenges with the directory-based organization.
+
+The aim of this TEP is to create a new organization contract for the Tekton Catalog, building on the prior work and 
+experiences from [TEP-0003][tep-0003]. In [TEP-0079: Tekton Catalog Support Tiers][tep-0079], we plan to add a Tekton 
+Official Catalog which should use this new organization contract - this proposal is a blocker for [TEP-0079][tep-0079].
+
+## Proposal
+
+We propose supporting a git-based organization of Tekton Catalogs where versions are expressed using commits and tags.
+
+### Flat Structure 
+
+In this organization contract, we'll use a flat structure instead of the directory structure. The example shown above
+with the two `Tasks` (argocd and golang-build) and one `Pipeline` (go-release) would be reorganized as such:
+
+```shell
+./task/
+  /argocd
+    /README.md
+    /argocd.yaml
+    /samples
+    /tests
+    /OWNERS
+  /golang-build
+    /README.md
+    /golang-build.yaml
+    /samples
+    /tests
+    /OWNERS
+./pipeline/
+  /go-release
+    /README.md
+    /go-release.yaml
+    /samples
+    /tests
+    /OWNERS
+```
```

Comment:
Would the structure of yaml remain same as in mentioned in [TEP-0003](https://github.com/tektoncd/community/blob/main/teps/0003-tekton-catalog-organization.md#example-of-a-resource-from-catalog) i.e. the resource will have annotations right ?